packages/block-library/src/navigation/view.js

```diff
@@ -177,6 +177,20 @@ const { state, actions } = store( 'core/navigation', {
 					focusableElements[ focusableElements.length - 1 ];
 			}
 		},
+		initNav() {
+			const { ref } = getElement();
+			callbacks.collapseNav( ref );
+			window.addEventListener( 'resize', () =>
+				callbacks.collapseNav( ref )
+			);
```

Comment:
This isn't particularly performant as I believe it's called repeatedly on each increment of the window.

Could you use `window.matchMedia`?

```js
const mediaQuery = window.matchMedia("(max-width: 600px)");

mediaQuery.addEventListener("change", (event) => {
  if (event.matches) {
    // run collapsing logic
  } else {
    // run expanding logic
  }
});
```

Reply:
Note `600` needs to come from `constants.js` if you go down that route.

Reply:
Done in 9de7b8cf0b

Reply:
Also added the cleanup in 68d000ad11